src/simularium/VisGeometry/GeometryStore.ts

```diff
@@ -0,0 +1,352 @@
+        });
+    }
+
+    public get registry(): Registry {
+        return this._registry;
+    }
+
+    public forEachMesh(iteratee: (geo: MeshGeometry) => void): void {
+        this._registry.forEach((value) => {
+            const { displayType } = value;
+            if (displayType !== GeometryDisplayType.PDB) {
+                const agentGeo = value as MeshGeometry;
+                iteratee(agentGeo);
+            }
+        });
+    }
+
+    public getGeoForAgentType(entryName: string): AgentGeometry | null {
+        const geo = this._registry.get(entryName);
+        return geo || null;
+    }
+
+    public cancelAll(): void {
+        // note that this leaves cancelled things in the registries.
+        // This should be called before the registries are cleared and probably
+        // only makes sense to do if they are indeed about to be cleared.
+
+        // don't process any queued requests
+        TaskQueue.stopAll();
+        // signal to cancel any pending geometries
+        this._registry.forEach((value) => {
+            value.geometry.cancelled = true;
+        });
+    }
+
+    public cacheLocalAssets(assets: { [key: string]: string }): void {
+        forEach(assets, (value, key) => {
+            const path = checkAndSanitizePath(key);
+            this._cachedAssets.set(path, value);
+        });
+    }
+
+    private getNewSphereGeometry(meshName: string): MeshLoadRequest {
+        return {
+            mesh: new Mesh(VisAgent.sphereGeometry),
+            cancelled: false,
+            instances: new InstancedMesh(VisAgent.sphereGeometry, meshName, 1),
+        };
+    }
+
+    private setGeometryInRegistry(
+        key: string,
+        geometry,
+        displayType: GeometryDisplayType
+    ) {
+        this._registry.set(key, {
+            geometry,
+            displayType,
+        });
+    }
+
+    /** Downloads a PDB from an external source */
+    private fetchPdb(url: string): Promise<PDBModel | undefined> {
+        const pdbModel = new PDBModel(url);
+        this.setGeometryInRegistry(url, pdbModel, GeometryDisplayType.PDB);
+        return fetch(url)
+            .then((response) => {
+                if (!response.ok) {
+                    throw new Error(
+                        `Failed to fetch ${pdbModel.filePath} from ${url}`
+                    );
+                }
+                return response.text();
+            })
+            .then((data) => {
+                if (pdbModel.cancelled) {
+                    this._registry.delete(url);
+                    return Promise.resolve(undefined);
+                }
+                pdbModel.parsePDBData(data);
+                const pdbEntry = this._registry.get(url);
+                if (pdbEntry && pdbEntry.geometry === pdbModel) {
+                    this.mlogger.info("Finished downloading pdb: ", url);
+
+                    return pdbModel;
+                } else {
+                    // TODO: what should happen here?
```

Comment:
Yeah, this wasn't my note. I can probably just get rid of the else block all together, but I wasn't sure if this TODO was something that would remind someone (probably @toloudis ) of work he wanted to do

Reply:
it's still a TODO ... I can't even remember exactly how it would get there... setGeometryInRegistry (line 151) seems like it would ensure that the if clause is satisfied.